crates/hyperdrive-math/src/long/max.rs

```diff
@@ -563,15 +562,25 @@ mod tests {
     /// `calculate_max_long`'s functionality. With this in mind, we provide
     /// `calculate_max_long` with a budget of `U256::MAX` to ensure that the two
     /// functions are equivalent.
-    #[ignore]
     #[tokio::test]
     async fn fuzz_calculate_max_long() -> Result<()> {
         let chain = TestChain::new().await?;
+        let alice = chain.alice().await?;
 
         // Fuzz the rust and solidity implementations against each other.
         let mut rng = thread_rng();
         for _ in 0..*FAST_FUZZ_RUNS {
-            let state = rng.gen::<State>();
+            // Snapshot the chain.
+            let id = chain.snapshot().await?;
+
+            // Gen a random state.
+            let mut state = rng.gen::<State>();
+
+            // Make sure maturity times are in the future.
+            let current_block_timestamp = alice.now().await?;
+            state.info.long_average_maturity_time += current_block_timestamp;
```

Comment:
I don't think this matters here since calculate_max_long doesn't know about the current block timestamp.